apps/champions/lib/champions/users.ex

```diff
@@ -118,4 +145,44 @@ defmodule Champions.Users do
   def calculate_experience_to_next_level(level) do
     Math.pow(100, 1 + level / 10) |> ceil()
   end
+
+  @doc """
+  Get a user's available AFK rewards, according to their AFK reward rates and the time since their last claim.
+  If more than 12 hours have passed since the last claim, the user will have accumulated the maximum amount of rewards.
+  """
+  def get_afk_rewards(user_id) do
+    user = Users.get_user(user_id)
+
+    user.afk_reward_rates
+    |> Enum.map(fn reward_rate ->
+      currency = Currencies.get_currency(reward_rate.currency_id)
+      amount = calculate_afk_rewards(user, reward_rate)
+      %{currency: currency, amount: amount}
+    end)
+  end
+
+  defp calculate_afk_rewards(user, afk_reward_rate) do
+    last_claim = user.last_afk_reward_claim
+    now = DateTime.utc_now()
+
+    minutes_since_last_claim = DateTime.diff(now, last_claim, :minute)
+    hours_since_last_claim = div(minutes_since_last_claim, 60)
```

Comment:
Isn't this unnecesarily costly? just check if minutes_since_last_claim > 12 * 60. You can set `@maximum_hours 12` or `@minutes_in_hour 60` if what you're looking for is clarity

Reply:
in fact you could just do min(minutes_since_last_claim, 12*60) and use that value